### PR TITLE
XP-209 create keras participant example

### PR DESCRIPTION
**Pull request by janpetschexain**
### References

- [XP-209](https://xainag.atlassian.net/browse/XP-209)

### Summary

- add an abstract `Participant` class to the sdk
- add a custom implementation for Keras/Tensorflow of the `Participant`

### Are there any open tasks/blockers for the ticket?

- documentation will be added with [XP-211](https://xainag.atlassian.net/browse/XP-211)

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: xain_sdk/examples/keras/participant.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+from typing import Dict, List, Tuple
+
+import numpy as np
+from tensorflow import Tensor
+from tensorflow.data import Dataset
+from tensorflow.keras import Input, Model
+from tensorflow.keras.layers import Dense
+
+import xain_sdk
```
Comment (atymoshchuk) [inline xain_sdk/examples/keras/participant.py]: Can you please import Participant from xain_sdk, as it's the only class we use, would be clear for an example that we use Participant from sdk and have to redefine some methods of it.

File: xain_sdk/sdk/participant.py [hunk 2]
```diff
@@ -12,3 +18,30 @@ def start(coordinator_url: str, use_case: UseCase):
...
+
+
+class Participant(ABC):
+    """An abstract participant for federated learning."""
+
+    @abstractmethod
+    def train_round(
+        self, weights: List[np.ndarray], epochs: int, epoch_base: int
+    ) -> Tuple[List[np.ndarray], Dict[str, List[np.ndarray]]]:
+        # pylint: disable=line-too-long
```
Comment (atymoshchuk) [inline xain_sdk/sdk/participant.py]: Would be nice if we can format it without disabling linter.

Comment (PanicButtonPressed) [reply xain_sdk/sdk/participant.py]: We disabled it sometimes for the return statement in order to have the types in one single line. Probably, we should have a look at the documentation, how it looks like later.

Comment (atymoshchuk) [reply xain_sdk/sdk/participant.py]: I will add it as a ticket for a next sprint, thanks.

File: xain_sdk/examples/keras/participant.py (hunk 1 above)
Comment (janpetschexain) [reply xain_sdk/examples/keras/participant.py]: done

**Pull request merged**